tests/testthat/test-independent-testing-prepare_ae_specificr.R

```diff
@@ -0,0 +1,340 @@
+                       population = adsl)
+
+plan_pilot <- plan |>
+  subset(pilot) |>
+  mutate(display_total = analysis %in% c("ae_specific"))
+
+meta_test <- meta_test |>
+  define_plan(plan_pilot)
+
+meta_test <- meta_test |>
+  define_population(name ="apat",
+                    group ="TRT01A",
+                    subset =ITTFL == "Y",
+                    label = "(All Participants as Treated)")
+
+meta_test <- meta_test |>  define_observation(name ="trtemfl",
+                                              group= "TRT01A",
+                                              var= "AEDECOD",
+                                              subset = TRTEMFL=="Y",
+                                              label= "Treatment emergent")
+
+meta_test <- meta_test |> define_parameter(name ="any")
+meta_test <- meta_test |>
+
+  define_analysis(name ="ae_specific",
+                  title="Participants With {term1} Adverse Events {term2}
+                         \\line (Incidence > 0% in One or More Treatment Groups)")
+
+
+meta_test <- meta_test %>% meta_build()
+
+meta_test$plan <- meta_test$plan |>
+  mutate(output_report = spec_filename(meta_test))
+
+test_meta_specific <- meta_test
+
+
+x_any <- prepare_ae_specific(test_meta_specific,
+                             population = "apat",
+                             observation = "trtemfl",
+                             parameter = "any")
+
+yy <- tibble(cbind(x_any$name, x_any$n)) %>% mutate(f = row_number())
+
+yyp <- x_any$prop %>% mutate(f = row_number())
+yypd <- x_any$diff %>% mutate(f = row_number())
+
+adsl_tot <- adsl %>% mutate(TRT01AN = 99) # add total rows into calculation
+adsl_tot <- rbind(adsl, adsl_tot)
+
+res_tot <- adsl_tot %>%
+  group_by(TRT01AN, ITTFL) %>%
+  summarise(n = n_distinct(USUBJID), .groups = "drop")
+
+res_tot1 <- res_tot %>%
+  mutate(z = n) %>%
+  select(-c(n))
+
+adae_itt <- full_join(adsl %>% select(USUBJID, TRT01AN, ITTFL), # merge with adsl to get percentage
+                      adae,
+                      by = "USUBJID"
+) %>% subset(ITTFL == "Y" & TRTEMFL == "Y")
+
+adae_tot <- adae_itt %>% mutate(TRT01AN = 99) # add total rows into calculation
+adae_tot <- rbind(adae_itt, adae_tot) %>% mutate(AEDECOD = metalite.ae:::to_sentence(AEDECOD), AEBODSYS = metalite.ae:::to_sentence(AEBODSYS))
+
+
+test_that("Group matching", {
+  expect_equal(x_any$group, c("Placebo", "Xanomeline Low Dose", "Xanomeline High Dose", "Total"))
+})
+
+test_that("population count", {
+  # Population count
+  res <- adsl_tot %>%
+    group_by(TRT01AN, ITTFL) %>%
+    summarise(n = n_distinct(USUBJID), .groups = "drop")
+
+  res <- res %>%
+    ungroup() %>%
+    mutate(x = paste0("n_", row_number()))
+
+  pop_cnt <- res %>%
+    pivot_wider(
+      id_cols = "ITTFL",
+      names_from = "x",
+      values_from = n,
+      values_fill = list(n = 0)
+    )
+  expect_equal(yy %>% filter(f == 1) %>% select(-c("x_any$name", f)), tibble(pop_cnt) %>% select(-c(ITTFL)))
```

Comment:
Hi @srnep , I only have a minor comment as I have in [Madhu's PR](https://github.com/Merck/metalite.ae/pull/69#discussion_r919421593). Could you please confirm if the derivation of `pop_cnt` is the same as that in SAS macro?